R/nm-tables.R

```diff
@@ -84,14 +84,12 @@ nm_tables <- function(
 #' @export
 nm_table_files <- function(.mod, .check_exists = TRUE) {
   .p <- get_model_path(.mod)
-  .l <- parse_ctl_to_list(.p)
+  .l <- nmrec::read_ctl(.p)
   out_dir <- get_output_dir(.mod, .check_exists = .check_exists)
 
   # get file names from table statements and construct paths
-  .f <- .l[grep("^TAB", names(.l))] %>%
-    map_chr(~paste(.x, collapse = " ")) %>%
-    str_extract("\\bFILE\\s*=\\s*([^ ]+)") %>%
-    str_replace("\\bFILE\\s*=\\s*", "") %>%
+  .f <- nmrec::select_records(.l, "table") %>%
+    purrr::map_chr(~nmrec::get_record_option(.x, "file")$value) %>%
```

Comment:
It's okay as is, and I know purrr's anonymous function syntax is used a lot in this code base, but using `function(...) ...` instead would allow `R CMD check` to flag/warn about issues inside the function.

Reply:
fixed